### PR TITLE
9 add toolbar

### LINKED ISSUE #9: Add toolbar for the pages, replace side nav
WCS had a toolbar, and it looked much nicer than what is currently in main. Discuss with group.

### PR DESCRIPTION
**Pull request by OllieWillette667**
Issue nine is replace the sidenav with a toolbar.

Closes #9

### CODE REVIEW

File: client/src/app/app-routing.module.ts
Comment (NicMcPhee) [inline client/src/app/app-routing.module.ts:10]: Nice catch.

File: client/cypress/integration/app.spec.ts
Comment (NicMcPhee) [inline client/cypress/integration/app.spec.ts:12]: Is there some kind of test that the tool bar exists and works?

File: client/src/app/app.component.html
Comment (NicMcPhee) [inline client/src/app/app.component.html]: I'm confused by the naming here. There's still lots of `sidenav`, but there's no side navigation bar, right?

File: client/src/styles.scss
Comment (NicMcPhee) [inline client/src/styles.scss]: I'd drop the "looks nice" comment – that really doesn't add anything useful.

Comment (NicMcPhee) [reply client/src/styles.scss]: Also, should we be using named colors (like `$primary`) from our CSS, i.e., creating a named color (that is just `white`) and referring to that name here?

**Review by NicMcPhee — COMMENTED** [review]
We should talk in the chat about these.

File: client/src/app/app.component.html
Comment (NicMcPhee) [reply client/src/app/app.component.html]: Maybe we want a `mat-toolbar` instead? https://material.angular.io/components/toolbar/overview

File: client/cypress/integration/app.spec.ts
Comment (jpwalbran) [inline client/cypress/integration/app.spec.ts]: It's probably a good idea to include a check that the navbar exists before trying to test the navigation. Something like cy.get(<toolbar-indicator>).should('have.text', 'My Pantry') or something similar. This could potentially save some hassle if it were to break in the future.

File: client/cypress/support/app.po.ts
Comment (jpwalbran) [inline client/cypress/support/app.po.ts]: We should have this data type allow for 'Shopping List' as well as pantry and products even if we don't use them in tests currently.

File: client/src/styles.scss
Comment (jpwalbran) [inline client/src/styles.scss]: Can we get rid of this color value? It seems like a hold over from the old code-base, and the color choice doesn't match with the rest of the current project.

File: client/src/styles.scss
Comment (jpwalbran) [inline client/src/styles.scss]: Again, can we avoid changing the colors here? (and revert the other changes to this file).

**Review by jpwalbran — CHANGES_REQUESTED** [review]
Looks like a good start, but there are a couple of things that I think should be changed here, mostly detailed elsewhere, except that currently there's still an HTML container and text with the old toolbar text (ie.HandyPantry -> MyPantry) that are at the bottom of each screen that should be removed.

File: client/cypress/integration/app.spec.ts
Comment (OllieWillette667) [reply client/cypress/integration/app.spec.ts]: Good idea, added.

File: client/src/styles.scss
Comment (OllieWillette667) [reply client/src/styles.scss]: Done

File: client/src/app/app.component.html
Comment (OllieWillette667) [reply client/src/app/app.component.html]: fixed

File: client/cypress/support/app.po.ts
Comment (OllieWillette667) [reply client/cypress/support/app.po.ts]: That seems like a different issue.

File: client/cypress/integration/app.spec.ts
Comment (NicMcPhee) [inline client/cypress/integration/app.spec.ts]: This link is weird, and doesn't really say anything except that there will be a URL there. Would it make more sense to say that the URL should _not_ contain the word "pantry" since we're using `/` as the pantry entry?

Also, would it be reasonable to include a check that clicking on "Shopping List" takes you to a URL containing "shoppinglist"?

**Review by NicMcPhee — CHANGES_REQUESTED** [review]
Sorry to keep dragging this out. It looks really good except for the one comment about testing the URLs. I'll try to bang out some changes there quick and commit them so we can get this done and in.

File: client/cypress/support/app.po.ts
Comment (NicMcPhee) [reply client/cypress/support/app.po.ts:6]: I agree that it could be a different issue, but if you want to take that position then you _definitely_ should create the issue. It's definitely weird that this doesn't have `'Shopping List'` and (perhaps more troubling) that there are no tests that seem to care?

Comment (NicMcPhee) [reply client/cypress/support/app.po.ts]: Actually, @OllieWillette667, I don't think it was a separate issue. I think that my comment below about filling out the tests in `app.spec.ts` would "force" the addition of `'Shopping List'` here. That seems to be the only place that `getNavLink()` is used, and if we make it a little more complete we'll end up addressing @jpwalbran's concern.

File: client/cypress/integration/app.spec.ts
Comment (NicMcPhee) [inline client/cypress/integration/app.spec.ts:12]: For some reason this `it` clause wasn't being run when it was nested inside the previous `it`. It wasn't clear why it was nested there in the first place, so I pulled it out and these tests are now running.

File: client/cypress/integration/app.spec.ts
Comment (NicMcPhee) [inline client/cypress/integration/app.spec.ts:18]: Added tests for the shopping list URL, and changed the pantry test to check that `'pantry'` does _not_ appear in the URL.

File: client/cypress/support/app.po.ts
Comment (NicMcPhee) [inline client/cypress/support/app.po.ts:7]: Add `'Shopping List'` to the set of `navOption`s so that we can test that part of the toolbar as well.

I also replaced `'${navOption}'` with the simpler (and in this case equivalent) `navOption`.

**Review by NicMcPhee — DISMISSED** [review]
I think this resolves all my outstanding issues. @OllieWillette667 should have a look and make sure my changes look OK, but feel free to merge this in if it looks good.

**Review by jpwalbran — APPROVED** [review]

File: client/cypress/support/app.po.ts
Comment (OllieWillette667) [reply client/cypress/support/app.po.ts:7]: Thank you for all the changes, I really appreciate you.

**Pull request merged**
